src/basic/util.c

```diff
@@ -90,7 +90,11 @@ bool in_initrd(void) {
         }
 
 check_all:
-        saved_in_initrd = path_is_temporary_fs("/") > 0;
+        r = path_is_temporary_fs("/");
+        if (r < 0)
+                log_error_errno(r, "Couldn't determine if / is a temporary file system: %m");
```

Comment:
in systemd functions must be in one of two categories: logging or non-logging. "library-like" functions generally don#t log on their own above LOG_DEBUG (leaving such logging it to their caller), "main function"-like functions do — about every error they encounter (so that the caller doesn't anymore). We generally request that all functions are clearly of one kind or the other, so that accidental duplicate logging or cases where logging is entirely missing won't happen. So far this was "library-like" function (like pretty much everything in src/basic/), there was no logging above LOG_DEBUG. You add that here now however.

Now, I sympathesize with your goal here, this probably should be logged somewhere. hence maybe downgrade this to log_debug_errno(). (debug logging is always ok) — if you do all these philosophical ideas about "library-like" and "main-function-like" don't apply

Reply:
Ah, got it. Sorry for skipping the code style lesson before making this PR, I just found it's well doc'ed in docs/CODING_STYLE.md. Will update this part.